genic/main.c

```diff
@@ -129,7 +129,12 @@ int main(int argc, char **argv)
     if(All2.MakeGlassGas || All2.MakeGlassCDM)
         glass_evolve(pm, 14, "powerspectrum-glass-tot", ICP, NumPartCDM+NumPartGas);
   }
-
+  
+  /*Write initial positions into ICP struct*/
+  for(int j=0; j<NumPartCDM; j++)
```

Comment:
I think you want NumPartCDM + NumPartGas here? Also does this need to be set for neutrino particles as well?